ResearchKit/Common/ORKBaseGraphView.h

```diff
@@ -0,0 +1,153 @@
+@protocol ORKBaseGraphViewDelegate <NSObject>
+
+@optional
+
+- (void)graphViewTouchesBegan:(ORKBaseGraphView *)graphView;
+
+- (void)graphView:(ORKBaseGraphView *)graphView touchesMovedToXPosition:(CGFloat)xPosition;
+
+- (void)graphViewTouchesEnded:(ORKBaseGraphView *)graphView;
+
+@end
+
+
+ORK_AVAILABLE_DECL
+@protocol ORKBaseGraphViewDataSource <NSObject>
+
+@required
+
+- (NSInteger)graphView:(ORKBaseGraphView *)graphView numberOfPointsInPlot:(NSInteger)plotIndex;
+
+- (ORKRangePoint *)graphView:(ORKBaseGraphView *)graphView plot:(NSInteger)plotIndex valueForPointAtIndex:(NSInteger)pointIndex;
+
+@optional
+
+- (NSInteger)numberOfPlotsInGraphView:(ORKBaseGraphView *)graphView;
+
+- (NSInteger)numberOfDivisionsInXAxisForGraphView:(ORKBaseGraphView *)graphView;
+
+- (CGFloat)maximumValueForGraphView:(ORKBaseGraphView *)graphView;
+
+- (CGFloat)minimumValueForGraphView:(ORKBaseGraphView *)graphView;
+
+- (NSString *)graphView:(ORKBaseGraphView *)graphView titleForXAxisAtIndex:(NSInteger)pointIndex;
+
+@end
+
+
+/**
+ *  IMPORTANT: THIS IS AN ABSTRACT CLASS. IT HOLDS PROPERTIES & METHODS COMMON TO CLASSES LIKE ORKLineGraphView & ORKDiscreteGraphView.
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKBaseGraphView : UIView
+
+@property (nonatomic, readonly) CGFloat minimumValue;
+
+@property (nonatomic, readonly) CGFloat maximumValue;
+
+@property (nonatomic, getter=isLandscapeMode) BOOL landscapeMode;
+
+@property (nonatomic) BOOL showsVerticalReferenceLines;
+
+@property (nonatomic, weak, nullable) id <ORKBaseGraphViewDelegate> delegate;
+
+@property (nonatomic, weak) IBOutlet id <ORKBaseGraphViewDataSource> dataSource;
+
+/* Appearance */
+
+@property (nonatomic, strong, nullable) UIColor *axisColor;
+
+@property (nonatomic, strong, nullable) UIColor *axisTitleColor;
+
+@property (nonatomic, strong, nullable) UIFont *axisTitleFont;
+
+@property (nonatomic, strong, nullable) UIColor *referenceLineColor;
+
+@property (nonatomic, strong, nullable) UIColor *scrubberThumbColor;
+
+@property (nonatomic, strong, nullable) UIColor *scrubberLineColor;
+
+@property (nonatomic, strong, nullable) UIPanGestureRecognizer *panGestureRecognizer;
+
+@property (nonatomic, strong, nullable) NSString *emptyText;
+
+/* Support for image icons as legends */
+
+@property (nonatomic, strong, nullable) UIImage *maximumValueImage;
+
+@property (nonatomic, strong, nullable) UIImage *minimumValueImage;
+
+- (void)sharedInit;
+
+- (NSInteger)numberOfPlots;
+
+- (NSInteger)numberOfPointsinPlot:(NSInteger)plotIndex;
+
+- (void)scrubReferenceLineForXPosition:(CGFloat)xPosition;
+
+- (void)setScrubberViewsHidden:(BOOL)hidden animated:(BOOL)animated;
+
+- (void)refreshGraph;
+
+@end
+
+
+ORK_CLASS_AVAILABLE
+@interface ORKRangePoint : NSObject
+
+@property (nonatomic) CGFloat maximumValue;
+
+@property (nonatomic) CGFloat minimumValue;
+
+@property (nonatomic, getter=isEmpty) BOOL empty;
+
+- (instancetype)initWithMinimumValue:(CGFloat)minValue maximumValue:(CGFloat)maxValue;
```

Comment:
Nit: let's use `minimumValue` and `maximumValue` as the argument names here, for coherence with the method name.